File: backend/app/Models/Leave.php
<?php

namespace App\Models;

use Carbon\Carbon;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Leave extends Model
{
    use HasFactory;

    protected $fillable = [
        'employee_id',
        'approved_by',
        'approved_date',
        'leave_type_id',
        'request_date', // Thêm vào
        'start_date',   // Thêm vào
        'end_date',     // Thêm vào
        'reason',       // Thêm vào
        'note',
        'status',
    ];

    protected $appends = ['duration'];

    public function employee(): BelongsTo
    {
        return $this->belongsTo(Employee::class, 'employee_id');
    }

    public function approved(): BelongsTo
    {
        return $this->belongsTo(Employee::class, 'approved_by');
    }

    public function leaveType(): BelongsTo
    {
        return $this->belongsTo(LeaveType::class, 'leave_type_id');
    }

    public function getDurationAttribute()
    {
        if ($this->start_date && $this->end_date) {
            $start = Carbon::parse($this->start_date);
            $end = Carbon::parse($this->end_date);
            // Tính số ngày nghỉ, bao gồm cả ngày bắt đầu và kết thúc
            return $start->diffInDays($end) + 1;
        }
        return null;
    }
}

File: hr_management_app/lib/models/leave.dart
class Leave {
  final int employeeId;
  final int approvedBy;
  final String? approvedDate;
  final int leaveTypeId;
  final String leaveDate;
  final String? note;
  final String status;
  final String createdAt;
  final String updatedAt;

  Leave({
    required this.employeeId,
    required this.approvedBy,
    this.approvedDate,
    required this.leaveTypeId,
    required this.leaveDate,
    this.note,
    required this.status,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Leave.fromJson(Map<String, dynamic> json) {
    return Leave(
      employeeId: json['employee_id'] ?? 0,
      approvedBy: json['approved_by'] ?? 0,
      approvedDate: json['approved_date'],
      leaveTypeId: json['leave_type_id'] ?? 0,
      leaveDate: json['leave_date'] ?? '',
      note: json['note'],
      status: json['status'] ?? '',
      createdAt: json['created_at'] ?? '',
      updatedAt: json['updated_at'] ?? '',
    );
  }
}